main/src/ca/uwaterloo/flix/api/lsp/provider/CompletionProvider.scala

```diff
@@ -232,8 +231,11 @@ object CompletionProvider {
     * Characters that constitute a word.
     * This is more permissive than the parser, but that's OK.
     */
-  private val isWordChar = Letters.LegalLetter ++ Letters.OperatorLetter ++
-    Letters.MathLetter ++ Letters.GreekLetter ++ CharPredicate("@/.")
+  private def isWordChar(c: Char) = c.isLetter || Lexer.isMathNameChar(c) || Lexer.isGreekNameChar(c) || Lexer.isUserOp(c).isDefined || (c match {
```

Comment:
Pattern match is non-exhaustive. It should have a default false case.

Reply:
The fact that this is not reported by the Scala compiler is a bug in scala.

Reply:
We should extract the pattern match into its own method.

Reply:
Moreover, I believe use of c.isLetter is wrong. It includes all kinds of "letters" that are not ascii.

Reply:
I extracted the match into another method 👍 I haven't had any issues with `isLetter` and it is used many times in the lexer. Do you have any examples where it's not working as intended?